holmes/plugins/toolsets/bash/bash_toolset.py

```diff
@@ -0,0 +1,303 @@
+                status=ToolResultStatus.ERROR,
+                data="Error: Bash executable or command not found. Ensure bash is installed and the command is valid.",
+                params=params,
+            )
+        except Exception as e:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data=f"Error executing command '{full_kubectl_command}': {str(e)}",
+                params=params,
+            )
+
+    def get_parameterized_one_liner(self, params: Dict[str, Any]) -> str:
+        return self._build_kubectl_command(params, "<pod_name>")
+
+
+class RunBashCommand(BaseBashTool):
+    def __init__(self, toolset: BaseBashExecutorToolset):
+        super().__init__(
+            name="run_bash_command",
+            description=(
+                "Executes a given bash command and returns its standard output, "
+                "standard error, and exit code."
+                "The command is executed via 'bash -c \"<command>\"'."
+                "Only some commands are allowed."
+            ),
+            parameters={
+                "command": ToolParameter(
+                    description="The bash command string to execute.",
+                    type="string",
+                    required=True,
+                ),
+                "timeout": ToolParameter(
+                    description=(
+                        "Optional timeout in seconds for the command execution. "
+                        "Defaults to 60s."
+                    ),
+                    type="integer",
+                    required=False,
+                ),
+            },
+            toolset=toolset,
+        )
+
+    def _invoke(self, params: Dict[str, Any]) -> StructuredToolResult:
+        command_str = params.get("command")
+        timeout = params.get("timeout", 60)
+
+        if not command_str:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error="The 'command' parameter is required and was not provided.",
+                params=params,
+            )
+
+        if not isinstance(command_str, str):
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"The 'command' parameter must be a string, got {type(command_str).__name__}.",
+                params=params,
+            )
+        try:
+            safe_command_str = make_command_safe(command_str, self.toolset.config)
+        except (argparse.ArgumentError, ValueError) as e:
+            logging.info(f"Refusing LLM tool call {command_str}", exc_info=True)
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Refusing to execute bash command. Only some commands are supported and this is likely because requested command is unsupported. Error: {str(e)}",
+                params=params,
+            )
+
+        try:
+            process = subprocess.run(
```

Comment:
Can we extract the call to subprocess and reuse in all invoke calls? I think that they are basicly the same right? 
same as the structure result etc...